src/public/entity_provider.rs

```diff
@@ -113,7 +113,7 @@ impl EntityProvider {
 
         match fetch_schema_result {
             Ok(get_schema_output) => {
-                let schema = Schema::from_str(&get_schema_output.schema)?;
+                let schema = Schema::from_str(&get_schema_output.schema).unwrap();
```

Comment:
and same here. 

For each of these errors, you'll need to create an `impl From<SourceError> for OutputError`. In this case, `SourceError` is the error that comes from `Schema::from_str` and `OutputError` is `ProviderError`. The function will show how to convert `SourceError` into an `OutputError`. Then the `?` operator will work and you won't need to make a change here. You can look at any `error.rs` for examples. I also recommend looking at the docs for the crate `thiserror`.

Reply:
You can also implement this functionality inline inside of `map_err`. If there's multiple spots where the same error conversion happens, prefer the first method. Otherwise, either approach is fine. You can see the `Schema::from_str` conversion you modified below for details on `map_error`. Also https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err

Reply:
In this case the source error is CedarSchemaError, which is a enum with three values, only one of which converts into a ProviderError. In this case do I add other error types to provider error (guessing not?), or something else?